scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -894,23 +894,32 @@ class TypedPipeFactory[T] private (@transient val next: NoStackAndThen[(FlowDef,
   override def sumByLocalKeys[K, V](implicit ev: T <:< (K, V), sg: Semigroup[V]) =
     andThen(_.sumByLocalKeys[K, V])
 
-  override def asPipe[U >: T](fieldNames: Fields)(implicit flowDef: FlowDef, mode: Mode, setter: TupleSetter[U]) =
+  override def asPipe[U >: T](fieldNames: Fields)(implicit flowDef: FlowDef, mode: Mode, setter: TupleSetter[U]) = {
     // unwrap in a loop, without recursing
-    unwrap(this).asPipe[U](fieldNames)(flowDef, mode, setter)
+    val (unwrapped, st) = unwrap(this, Array())
+    val pipe = unwrapped.asPipe[U](fieldNames)(flowDef, mode, setter)
+    RichPipe.setPipeDescriptionFrom(pipe, LineNumber.tryNonScaldingCaller(st))
+    pipe
+  }
 
   override def toIterableExecution: Execution[Iterable[T]] = Execution.getConfigMode.flatMap {
     case (conf, mode) =>
       // This can only terminate in TypedPipeInst, which will
       // keep the reference to this flowDef
       val flowDef = new FlowDef
-      val nextPipe = unwrap(this)(flowDef, mode)
+      val (nextPipe, stackTraces) = unwrap(this, Array())(flowDef, mode)
       nextPipe.toIterableExecution
   }
 
   @annotation.tailrec
-  private def unwrap(pipe: TypedPipe[T])(implicit flowDef: FlowDef, mode: Mode): TypedPipe[T] = pipe match {
-    case TypedPipeFactory(n) => unwrap(n(flowDef, mode))
-    case tp => tp
+  private def unwrap(pipe: TypedPipe[T], st: Array[StackTraceElement])(implicit flowDef: FlowDef, mode: Mode): (TypedPipe[T], Array[StackTraceElement]) = pipe match {
+    case TypedPipeFactory(n) =>
+      val fullTrace = n match {
+        case NoStackAndThen.WithStackTrace(_, st) => st
+        case _ => Array[StackTraceElement]()
+      }
+      unwrap(n(flowDef, mode), st ++ fullTrace)
```

Comment:
We could get a line number out of each, possibly getting several line numbers. That might be better really, but it could appear more noisey. This was just attempting to do a priority of the first thing you did to the last, but thats pretty arbitrary